### PR TITLE
"design guide" page for admin

### PR DESCRIPTION
**Pull request by weedySeaDragon**
### PT Story:  Design standards  (prelim. to Style guide)
https://www.pivotaltracker.com/story/show/164809369

### Changes proposed in this pull request:
1.  Created a "design guide page" (have to be logged in as admin) so we have a place to see, discuss, and maybe record design decisions.
    - after logging is as admin, go to <localhost shf url>/admin/designguide
    - it's rather hard coded.  Not sure that we ever should put this into production, but it gives
       us something to start with.  (This really helped me when working thru the NewLook PRs)
2. Converted _nearly_ all the stylesheets to `.scss` (SASS)  We should be using .scss so that
     we can DRY up the CSS and take advantage of the power of SASS.    _(Should be a PT story)_
    Needed to do this to be able to pull out all color hex codes into variables, which is what I used to get a list of all colors in use.  (Using variables for colors is one good example of using variables in CSS.  You can look at how Boostrap uses variables, for example.  That's what lets everyone just define their own color for $primary, etc., and have it be put into all of the bootstrap stuff.)
3. Found out the hard way that ImagesUtility _requires_ `.css` files, so the 2 files it requires are not converted to .scss

### Screenshots (Optional):

designguide-toc

designguide-colors

**...**

---

designguide-btbs

designguide-btns-body

designguide-alerts
designguide-text

### Ready for review:
@thesuss @patmbolger @rmzse

### CODE REVIEW

**Comment by thesuss** [comment]
image

**Comment by thesuss** [comment]
image

**Comment by thesuss** [comment]
*wonders* is companies.scss even used anywhere anymore? 
I dont think so: 
```
@import "bootstrap";
@import "bootstrap-toggle";
@import "font-awesome-sprockets";
@import "font-awesome";
@import "maps";
@import "pagination";
@import "style";
@import "custom";
@import "dashboard";
```

**Comment by thesuss** [comment]
May be Different PR, just want to have it noted
image

File: app/views/admin_only/design_guide/show.html.haml [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+
+  .field-label {
+    font-weight: bold;
+  }
+
+  .code {
+  }
+
+
+%div#design-guide{ class: "#{controller.action_name}" }
```
[the remark this answers is not in the document]
Comment (thesuss) [reply app/views/admin_only/design_guide/show.html.haml]: Agree

**Comment by thesuss** [comment]
I was afraid to delete the page specific scss-files as I was unsure how/where they are called/used. But I dont think they are called/used.

**Comment by patmbolger** [comment]
I really like this idea _and_ the scss cleanup.

NOTE: All bottom buttons should be centered (your example showing a right-aligned button is out of date, I believe).

**Comment by patmbolger** [comment]
@weedySeaDragon Everything looks good except for the examples showing "button at the bottom of a page".

It looks like this:

---
Screen Shot 2019-04-10 at 4 08 34 PM

---

But the last 2 (of 3) of the examples are obsolete, since we standardized in _centering_ all buttons at the bottom of a page.  Examples:

---
Screen Shot 2019-04-10 at 4 08 54 PM

---

Screen Shot 2019-04-10 at 4 09 04 PM

**Review by patmbolger — CHANGES_REQUESTED** [review]
pls see comment

**Review by patmbolger — APPROVED** [review]

**Pull request merged**